Add integration test of scenario "first release"

mathias-baumann-sociomantic opened the pull request:
This also puts in place the infrastructure for future integration test
scenarios which can now easily be integrated in the existing
structure.

Depends on ~~https://github.com/sociomantic-tsunami/neptune/pull/83~~

mathias-baumann-sociomantic commented:
I don't trust travis.

leandro-lucarella-sociomantic commented:
Marking as blocked until #83 is merged.

mathias-baumann-sociomantic commented:
This is ready for review now as well, after lots of fights with travis

leandro-lucarella-sociomantic commented on test/initial_release/GHTestServer.d:
```diff
@@ -0,0 +1,200 @@
+
+*******************************************************************************/
+
+module test.initial_release.GHTestServer;
```
It might be a good idea to move to makd v2 or at least use `integrationtest` as the directory to store integration tests instead.

leandro-lucarella-sociomantic commented on test/initial_release/GHTestServer.d:
```diff
@@ -0,0 +1,200 @@
+/*******************************************************************************
+
+    Github Test Server class
```
I guess at some point this file should go to `test/lib` or something to be reusable from other tests, right?

mathias-baumann-sociomantic replied:
Yes, as soon as  a second test will be added I plan to adjust the structure accordingly

leandro-lucarella-sociomantic commented on test/initial_release/main.d:
```diff
@@ -0,0 +1,172 @@
+    import std.string : strip;
+    import std.algorithm : startsWith;
+
+    sleep(200.msecs);
```
The 10s times before looks OK as a safety net, but a 200ms timer in a test screams flakeyness in my eyes. Are you sure this is a good idea?

leandro-lucarella-sociomantic commented on test/initial_release/main.d:
```diff
@@ -0,0 +1,172 @@
+
+    auto file = File(format("%s/relnotes.md", data), "r");
+    auto fsize = file.size();
+    assert(fsize < 1024 * 16, "relnotes file unexpectedly large!");
```
Oh, now I miss Ocean's `UnitTestRunner` and the `Test` module... :cry:

mathias-baumann-sociomantic replied:
You mean this line triggered your sudden feeling of loss or more the whole module?

leandro-lucarella-sociomantic replied:
The whole module. Seeing an `assert` in a unittest looks really like a wasted opportunity :)

leandro-lucarella-sociomantic commented:
Trying this PR to release beaver v0.2.x -> v0.2.2 I get this:
```
beaver:v0.2.x %= ✔ $ neptune-release 
core.exception.RangeError@./submodules/octod/src/octod/core.d(143): Range violation
----------------
??:? _d_arraybounds [0x98b915]
??:? octod.core.__array [0x8d5c81]
./submodules/octod/src/octod/core.d:143 void octod.core.HTTPConnection.connect() [0x8cd5a9]
./submodules/octod/src/octod/core.d:88 octod.core.HTTPConnection octod.core.HTTPConnection.connect(octod.core.Configuration) [0x8cd327]
/home/luca/devel/neptune/src/release/main.d:76 _Dmain [0x7c57ca]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x992207]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x99212f]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x9921ac]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x99212f]
??:? _d_run_main [0x99209b]
??:? main [0x7d1ca1]
??:? __libc_start_main [0x45b3e82f]
```

I will leave it unreleased so it's easier to debug.

mathias-baumann-sociomantic commented on Build.mak:
```diff
@@ -9,20 +9,29 @@ override DFLAGS += -I./submodules/d-yaml/source -I./submodules/tinyendian/source
 # time for actual apps in this repo.
-$O/libvibed.a: $(shell find $C/submodules/vibed/source -type f -name '*.d')
+$O/libvibed.a: $(shell find $C/submodules/vibed/source -type f -name '*.d' \
+		-and -not -name 'appmain.d')
```
It's more a left-over for when I wanted to use vibed's main() function. I can remove it.

leandro-lucarella-sociomantic replied:
Yes, either removing it if not necessary or explaining why it exists would be acceptable for mw.

mathias-baumann-sociomantic commented on src/release/api.d:
```diff
@@ -44,26 +44,32 @@ public Configuration getConf ( )
 *******************************************************************************/
 
-public void checkOAuthSetup ( )
+public void checkOAuthSetup ( bool assume_yes )
```
Hmm I am not sure I like the side-effect in `readYesNoResponse()`.. it could come very unexpected to a new reader/user of the code. Maybe it could be an optional parameter for the function

leandro-lucarella-sociomantic replied:
For me it is perfectly acceptable and expected that an interactive function behaves differently given global configuration (because of that, this is what I did in the [git-hub tool](https://github.com/sociomantic-tsunami/git-hub/blob/v1.x.x/git-hub#L127-L171), in case you want to check how confusing, or not, is it).

There are a few differences, looking at it, like there is no *force-yes* but a it will *do nothing* if is not running in a interactive console, but it is similar in spirit.

But my feeling is, every global configuration will always change how functions behave in a way that is not "expected". The whole point of global config is convenience, and I think by using the global config in that `readYesNoResponse()` it makes all much more convenient. Maybe you can rename it as `chose()` or something, maybe the `read` part is what sounds weird if it won't read anything in some circumstances. But if it's just `chose()` or `ask()` for me it makes more sense that that choice can be made by the program itself in some situations.

mathias-baumann-sociomantic commented:
@leandro-lucarella-sociomantic this is the source of your error: https://github.com/sociomantic-tsunami/octod/pull/28

leandro-lucarella-sociomantic commented:
> @leandro-lucarella-sociomantic this is the fix for your error: sociomantic-tsunami/octod#28
> 
> Setting to blocked until that PR is merged and this one updated

merged

mathias-baumann-sociomantic commented on src/release/main.d:
```diff
@@ -203,11 +203,20 @@ void main ( string[] params )
 
+    if (opts.save_mail)
+    {
+        scope file = File("mail.txt", "w");
```
well, reading it from STDOUT requires me to actuall parse stdout.. which seemed a bit dirty and hacky to me..

mathias-baumann-sociomantic replied:
well, reading it from STDOUT requires me to actuall parse stdout.. which seemed a bit dirty and hacky to me..

leandro-lucarella-sociomantic replied:
How is it different from parsing a file?

mathias-baumann-sociomantic commented:
* Removed `save-email` option
* updated to makd 2
* made options globally accessible
* renamed `readYesNoResponse` to `getBoolChoice`
* `getBoolChoice` respects global `assume_yes` option
* removed 200.msecs delay in test
* updated octod with fixed version for port parsing
* removed unneeded lines from `Build.mak`
* fixed target dependencies in `Build.mak`

mihails-strasuns-sociomantic commented on integrationtest/initial_release/main.d:
```diff
@@ -0,0 +1,182 @@
+
+    auto tmp = readRequiredOption!string("tmp", "Temporary directory");
+    auto bin = readRequiredOption!string("bin", "Binary directory");
+    auto data = __FILE_FULL_PATH__[0 .. $-"main.d".length] ~ "data";
```
Ha ha, so one can't move project folder without all tests starting to fail? :)

mathias-baumann-sociomantic replied:
I don't understand? The path is always the current file, so moving it also updates the variable... of course you need to recompile, yes.

mihails-strasuns-sociomantic commented on integrationtest/initial_release/main.d:
```diff
@@ -0,0 +1,182 @@
+
+    scope(failure)
+    {
+        writefln("An assert failed. Neptune output was:\n------\n%s\n-----\n%s",
```
Not necessarily an assert

mihails-strasuns-sociomantic commented on integrationtest/initial_release/shellHelper.d:
```diff
@@ -0,0 +1,39 @@
+    Runs a command and returns the output
+
+    Params:
+        pwd = directory to run the command in
```
why `p`?

mathias-baumann-sociomantic replied:
I guess because I am used to the shell command `pwd` which stands for `print working directory`

mathias-baumann-sociomantic commented:
* Renamed pwd -> wd
* added comments / restructured asserts
* rephrased scope(failure) msg

nemanja-boric-sociomantic commented:
Nitpick: `cwd` is a more common naming.

mathias-baumann-sociomantic commented:
but it isn't the *current* working directory. It is the working directory that the given cmd should be run with

nemanja-boric-sociomantic commented:
Good point.

leandro-lucarella-sociomantic commented on Build.mak:
```diff
@@ -19,6 +19,8 @@ $O/libdyaml.a: $(shell find $C/submodules/d-yaml/source -type f -name '*.d') \
 
 ################################################################################
 
+$O/allunittests $O/fastunittest: $O/libvibed.a $O/libdyaml.a
```
You forgot to remove this line:
```make
%unittests: $O/libvibed.a $O/libdyaml.a
```

mathias-baumann-sociomantic replied:
Well, it is removed later. Do you want that removal to be moved to another commit?

leandro-lucarella-sociomantic commented on src/release/main.d:
```diff
@@ -21,8 +21,10 @@ import release.mergeHelper;
 import semver.Version;
 
 import octod.api.repos;
+import octod.api.releases;
```
I guess this change sneaked in this commit accidentally?

mathias-baumann-sociomantic replied:
No. The new way makd compiles my main.d file now makes dmd miss that import and that leads to linker errors. Importing it not as nested but instead on file scope fixes that

leandro-lucarella-sociomantic commented:
I have `git config commit.gpgSign` set to `true` and I'm getting this error while trying to run the integration tests:
```
   run build/devel/tmp/test-initial_release
Listening for requests on http://[::]:8080/
Failed to listen on 0.0.0.0:8080
Task terminated with uncaught exception: error: gpg failed to sign the data
fatal: failed to write commit object

core.exception.AssertError@/home/luca/devel/neptune/integrationtest/initial_release/main.d(52): Assertion failure
----------------
??:? _d_assert [0xa1137c]
??:? void integrationtest.initial_release.main.__assert(int) [0x849151]
/home/luca/devel/neptune/integrationtest/initial_release/main.d:52 _Dmain [0x845255]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0xa17d33]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0xa17c5b]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0xa17cd8]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0xa17c5b]
??:? _d_run_main [0xa17bc7]
??:? main [0x8491a9]
??:? __libc_start_main [0x3ceb282f]
```

leandro-lucarella-sociomantic commented on Build.mak:
```diff
@@ -19,10 +19,14 @@ $O/libdyaml.a: $(shell find $C/submodules/d-yaml/source -type f -name '*.d') \
 
-%unittests: $O/libvibed.a $O/libdyaml.a
+ITFLAGS += --tmp=$O --bin=$B
+integrationtest: $B/neptune-release $O/libvibed.a
```
To be strictly correct, this should also be set per `$O/test-*`, which is quite annoying. We could leave it like this as in practice it will only fail if someone tries to build one individual test, but if you want to do it right, you could do something like:

```make
$(patsubst $T/$(INTEGRATIONTEST)/%/main.d,$O/test-%.stamp,\
		$(filter-out $(TEST_FILTER_OUT),\
		$(wildcard $T/$(INTEGRATIONTEST)/*/main.d))): $B/neptune-release $O/libvibed.a
```

This might be something to add to MakD. Feel free to create an issue there.

For now you could also just list the tests manually, it might be simpler and more clear:
```make
$O/test-initial_release: $B/neptune-release $O/libvibed.a
```

mathias-baumann-sociomantic replied:
Update: 
* made sure test turns of gpg
* updated Build.mak

mathias-baumann-sociomantic commented:
Update:

made sure test turns of gpg

leandro-lucarella-sociomantic commented:
Needs rebasing

leandro-lucarella-sociomantic commented:
(I think is a good time to rebase all your PRs to the current master)

mathias-baumann-sociomantic commented:
rebased

leandro-lucarella-sociomantic commented on Build.mak:
```diff
@@ -24,7 +24,9 @@ $O/allunittests $O/fastunittest: $O/libvibed.a $O/libdyaml.a
 $O/%unittests: override LDFLAGS += -L$O -lvibed -ldyaml -levent -lssl -lcrypto
 $O/pkg-neptune.stamp: release overview
 
-%unittests: $O/libvibed.a $O/libdyaml.a
```
It is quite confusing in this change that some unrelated dependencies are removed, specially when in the diff context you can see they are already there, so I would separate this small change in a different commit just for clarity.

leandro-lucarella-sociomantic commented:
Besides that small last comment, I'm getting this when running the tests locally:

```
   run build/devel/tmp/test-initial_release
Listening for requests on http://[::]:8080/
Failed to listen on 0.0.0.0:8080
Task terminated with unhandled exception: Assertion failure
Failure! Neptune output was:
------
[97A63FA8:00000000 dia] libevent version: 2.0.21-stable
[97A63FA8:00000000 dia] libevent is using epoll for events.
[97A63FA8:00000000 trc] Connecting to GitHub API server ...
[97A63FA8:00000000 dbg] Create HTTP client pool 127.0.0.1:8080 false proxy :0
[97A63FA8:00000000 dbg] creating new HTTPClient connection, all 0 are in use
[97A63FA8:00000000 dbg]  ... 7FA897A65100
[97A63FA8:00000000 dbg] Now got 1 connections
[97A63FA8:00000000 trc] Connected.
[97A63FA8:00000000 trc] GET /repos/tester/sandbox
[97A63FA8:00000000 dbg] dnsresolve 127.0.0.1
[97A63FA8:00000000 dbg] dnsresolve ret
[97A63FA8:00000000 dbg] Socket event on fd 9: 128 (16570F8 vs 16570F8)
[97A63FA8:00000000 trc] Connect result status: 128
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] HTTP client request:
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] GET http://127.0.0.1:8080/repos/tester/sandbox HTTP/1.1
[97A63FA8:00000000 trc] User-Agent: vibe.d/0.7.32 (HTTPClient, +http://vibed.org/)
[97A63FA8:00000000 trc] Connection: keep-alive
[97A63FA8:00000000 trc] Accept-Encoding: gzip, deflate
[97A63FA8:00000000 trc] Host: 127.0.0.1
[97A63FA8:00000000 trc] Authorization: bearer 0000
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 256 B
[97A63FA8:00000000 trc] Accept: application/vnd.github.v3+json
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 4 B
[97A63FA8:00000000 trc] HTTP client waiting for response
[97A63FA8:00000000 trc] socket 9 write event (false)!
[97A63FA8:00000000 trc] socket 9 read event!
[97A63FA8:00000000 trc] HTTP client reading status line
[97A63FA8:00000000 trc] stln: HTTP/1.1 200 OK
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP client response:
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP/1.1 200 OK
[97A63FA8:00000000 trc] Server: vibe.d/0.7.32
[97A63FA8:00000000 trc] Date: Tue, 26 Sep 2017 13:34:37 GMT
[97A63FA8:00000000 trc] Keep-Alive: timeout=10
[97A63FA8:00000000 trc] Content-Type: application/json; charset=UTF-8
[97A63FA8:00000000 trc] Content-Length: 45
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] GET /repos/tester/sandbox/releases
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] HTTP client request:
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] GET http://127.0.0.1:8080/repos/tester/sandbox/releases HTTP/1.1
[97A63FA8:00000000 trc] User-Agent: vibe.d/0.7.32 (HTTPClient, +http://vibed.org/)
[97A63FA8:00000000 trc] Connection: keep-alive
[97A63FA8:00000000 trc] Accept-Encoding: gzip, deflate
[97A63FA8:00000000 trc] Host: 127.0.0.1
[97A63FA8:00000000 trc] Authorization: bearer 0000
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 256 B
[97A63FA8:00000000 trc] Accept: application/vnd.github.v3+json
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 13 B
[97A63FA8:00000000 trc] HTTP client waiting for response
[97A63FA8:00000000 trc] socket 9 write event (false)!
[97A63FA8:00000000 trc] socket 9 read event!
[97A63FA8:00000000 trc] HTTP client reading status line
[97A63FA8:00000000 trc] stln: HTTP/1.1 200 OK
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP client response:
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP/1.1 200 OK
[97A63FA8:00000000 trc] Server: vibe.d/0.7.32
[97A63FA8:00000000 trc] Date: Tue, 26 Sep 2017 13:34:37 GMT
[97A63FA8:00000000 trc] Keep-Alive: timeout=10
[97A63FA8:00000000 trc] Content-Type: application/json; charset=UTF-8
[97A63FA8:00000000 trc] Content-Length: 2
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] GET /repos/tester/sandbox/tags
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] HTTP client request:
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] GET http://127.0.0.1:8080/repos/tester/sandbox/tags HTTP/1.1
[97A63FA8:00000000 trc] User-Agent: vibe.d/0.7.32 (HTTPClient, +http://vibed.org/)
[97A63FA8:00000000 trc] Connection: keep-alive
[97A63FA8:00000000 trc] Accept-Encoding: gzip, deflate
[97A63FA8:00000000 trc] Host: 127.0.0.1
[97A63FA8:00000000 trc] Authorization: bearer 0000
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 256 B
[97A63FA8:00000000 trc] Accept: application/vnd.github.v3+json
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 9 B
[97A63FA8:00000000 trc] HTTP client waiting for response
[97A63FA8:00000000 trc] socket 9 write event (false)!
[97A63FA8:00000000 trc] socket 9 read event!
[97A63FA8:00000000 trc] HTTP client reading status line
[97A63FA8:00000000 trc] stln: HTTP/1.1 200 OK
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP client response:
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP/1.1 200 OK
[97A63FA8:00000000 trc] Server: vibe.d/0.7.32
[97A63FA8:00000000 trc] Date: Tue, 26 Sep 2017 13:34:37 GMT
[97A63FA8:00000000 trc] Keep-Alive: timeout=10
[97A63FA8:00000000 trc] Content-Type: application/json; charset=UTF-8
[97A63FA8:00000000 trc] Content-Length: 2
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] GET /repos/tester/sandbox/milestones?state=all
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] HTTP client request:
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] GET http://127.0.0.1:8080/repos/tester/sandbox/milestones?state=all HTTP/1.1
[97A63FA8:00000000 trc] User-Agent: vibe.d/0.7.32 (HTTPClient, +http://vibed.org/)
[97A63FA8:00000000 trc] Connection: keep-alive
[97A63FA8:00000000 trc] Accept-Encoding: gzip, deflate
[97A63FA8:00000000 trc] Host: 127.0.0.1
[97A63FA8:00000000 trc] Authorization: bearer 0000
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 256 B
[97A63FA8:00000000 trc] Accept: application/vnd.github.v3+json
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 25 B
[97A63FA8:00000000 trc] HTTP client waiting for response
[97A63FA8:00000000 trc] socket 9 write event (false)!
[97A63FA8:00000000 trc] socket 9 read event!
[97A63FA8:00000000 trc] HTTP client reading status line
[97A63FA8:00000000 trc] stln: HTTP/1.1 200 OK
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP client response:
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP/1.1 200 OK
[97A63FA8:00000000 trc] Server: vibe.d/0.7.32
[97A63FA8:00000000 trc] Date: Tue, 26 Sep 2017 13:34:38 GMT
[97A63FA8:00000000 trc] Keep-Alive: timeout=10
[97A63FA8:00000000 trc] Content-Type: application/json; charset=UTF-8
[97A63FA8:00000000 trc] Content-Length: 2
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] GET /repos/tester/sandbox/branches
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] HTTP client request:
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] GET http://127.0.0.1:8080/repos/tester/sandbox/branches HTTP/1.1
[97A63FA8:00000000 trc] User-Agent: vibe.d/0.7.32 (HTTPClient, +http://vibed.org/)
[97A63FA8:00000000 trc] Connection: keep-alive
[97A63FA8:00000000 trc] Accept-Encoding: gzip, deflate
[97A63FA8:00000000 trc] Host: 127.0.0.1
[97A63FA8:00000000 trc] Authorization: bearer 0000
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 256 B
[97A63FA8:00000000 trc] Accept: application/vnd.github.v3+json
[97A63FA8:00000000 trc] --------------------
[97A63FA8:00000000 trc] evbuffer_add (fd 9): 13 B
[97A63FA8:00000000 trc] HTTP client waiting for response
[97A63FA8:00000000 trc] socket 9 write event (false)!
[97A63FA8:00000000 trc] socket 9 read event!
[97A63FA8:00000000 trc] HTTP client reading status line
[97A63FA8:00000000 trc] stln: HTTP/1.1 200 OK
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP client response:
[97A63FA8:00000000 trc] ---------------------
[97A63FA8:00000000 trc] HTTP/1.1 200 OK
[97A63FA8:00000000 trc] Server: vibe.d/0.7.32
[97A63FA8:00000000 trc] Date: Tue, 26 Sep 2017 13:34:38 GMT
[97A63FA8:00000000 trc] Keep-Alive: timeout=10
[97A63FA8:00000000 trc] Content-Type: application/json; charset=UTF-8
[97A63FA8:00core.exception.AssertError@/home/luca/devel/neptune/integrationtest/initial_release/main.d(158): Assertion failure
----------------
??:? _d_assert [0xa113b4]
??:? void integrationtest.initial_release.main.__assert(int) [0x849189]
/home/luca/devel/neptune/integrationtest/initial_release/main.d:158 void integrationtest.initial_release.main.runNeptune(integrationtest.initial_release.GHTestServer.RestAPI) [0x845967]
/usr/include/dmd/phobos/std/functional.d-mixin-1216:1227 void std.functional.DelegateFaker!(void function(integrationtest.initial_release.GHTestServer.RestAPI)*).DelegateFaker.doIt(integrationtest.initial_release.GHTestServer.RestAPI) [0x92507b]
./submodules/vibed/source/vibe/core/core.d:595 void vibe.core.core.makeTaskFuncInfo!(void delegate(integrationtest.initial_release.GHTestServer.RestAPI), integrationtest.initial_release.GHTestServer.RestAPI).makeTaskFuncInfo(ref void delegate(integrationtest.initial_release.GHTestServer.RestAPI), ref integrationtest.initial_release.GHTestServer.RestAPI).callDelegate(vibe.core.core.TaskFuncInfo*) [0x9b30aa]
./submodules/vibed/source/vibe/core/core.d:1224 void vibe.core.core.CoreTask.run() [0x9af6e2]
??:? void core.thread.Fiber.run() [0xa6c3f7]
??:? fiber_entryPoint [0xa6c15a]
??:? [0xffffffff]
000000 trc] Content-Length: 2
[97A63FA8:00000000 trc] ---------------------
release.shellHelper.ExitCodeException@./src/release/shellHelper.d(55): Cmd Failed: fatal: Not a valid object name: 'v1.0.0'.
 (status 128)
----------------
./src/release/shellHelper.d:55 immutable(char)[] release.shellHelper.cmd(immutable(char)[]) [0x7e23d0]
./src/release/actions.d:141 immutable(char)[] release.actions.LocalAction.execute() [0x8baf9f]
/home/luca/devel/neptune/src/release/main.d:127 _Dmain [0x7c5c71]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0x99194f]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x991877]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0x9918f4]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0x991877]
??:? _d_run_main [0x9917e3]
??:? main [0x7d19d9]
??:? __libc_start_main [0x9667782f]
[97A63FA8:00000000 dia] Main thread exiting

-----
Warning: No previous releases found. This should only be the case for your very first release!
We are on branch v1.x.x
Detected release v1.0.0
Warning: No corresponding milestone found for v1.0.0
Actions to be done:
 * Creating annotated tag v1.0.0 (git tag -F- v1.0.0 v1.x.x)
 * Create tracking branch v1.0.x (git branch v1.0.x v1.0.0)
 * Create next major branch v2.x.x (git branch v2.x.x)
 * Checkout next major branch v2.x.x (git checkout v2.x.x)
 * Removing release notes (git rm relnotes/*.md)
 * Commiting removal of release notes (git commit -m "Clear release notes after release")
 * Checkout original branch v1.x.x (git checkout v1.x.x)

Creating annotated tag v1.0.0 ...
	git tag -F- v1.0.0 v1.x.x
	
Create tracking branch v1.0.x ...
	git branch v1.0.x v1.0.0

submodules/makd/Makd.mak:615: recipe for target '/home/luca/devel/neptune/build/devel/tmp/test-initial_release.stamp' failed
make: *** [/home/luca/devel/neptune/build/devel/tmp/test-initial_release.stamp] Error 1
```

mathias-baumann-sociomantic commented:
I wish you'd stop breaking my stuff with your weird system.

* Put makefile chaneg in extra commit
* Probably fixed your problem, please try again

leandro-lucarella-sociomantic commented:
> Probably fixed your problem, please try again

NOPE. The error is definitely shorter though. I guess that's a good thing :thinking: 

```
   run build/devel/tmp/test-initial_release
Listening for requests on http://[::]:8080/
Failed to listen on 0.0.0.0:8080
Task terminated with uncaught exception: error: gpg failed to sign the data
fatal: failed to write commit object

core.exception.AssertError@/home/luca/devel/neptune/integrationtest/initial_release/main.d(52): Assertion failure
----------------
??:? _d_assert [0xa11494]
??:? void integrationtest.initial_release.main.__assert(int) [0x849269]
/home/luca/devel/neptune/integrationtest/initial_release/main.d:52 _Dmain [0x845295]
??:? _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZ9__lambda1MFZv [0xa17e4b]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0xa17d73]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).runAll() [0xa17df0]
??:? void rt.dmain2._d_run_main(int, char**, extern (C) int function(char[][])*).tryExec(scope void delegate()) [0xa17d73]
??:? _d_run_main [0xa17cdf]
??:? main [0x8492c1]
??:? __libc_start_main [0x6fac82f]
make: *** [/home/luca/devel/neptune/build/devel/tmp/test-initial_release.stamp] Error 1
submodules/makd/Makd.mak:615: recipe for target '/home/luca/devel/neptune/build/devel/tmp/test-initial_release.stamp' failed
```

leandro-lucarella-sociomantic commented:
BTW, asserts suck for testing, as you don't get what actually went wrong. I think it might make sense to move to ocean's test sooner than later (or look for D2 native alternatives if you prefer, but I think what we have in ocean is pretty solid).

mathias-baumann-sociomantic commented:
Have you tried a `make clean` before?

mathias-baumann-sociomantic commented:
Is your git commit command in anyway aliased or so so that the shell doesn't run the pure `git commit`?
Because your global options should no longer be applied as I overwrite your `$HOME` variable

mathias-baumann-sociomantic commented:
I think I found the problem and have rectified

leandro-lucarella-sociomantic commented:
I believe I did a `make clean`, but I can try again if it helps you.

leandro-lucarella-sociomantic commented:
I just did the `make clean` and I still get the error. BTW, I'm using `dmd-bin=2.073.2-0`. I'm actually using [beaver's make wrapper](https://github.com/sociomantic-tsunami/beaver/blob/f3a4dd0d6f7932af463aa573ec2318728ed1b832/tools/make-wrapper-dlang) to build using docker as close to the travis environment as possible.

leandro-lucarella-sociomantic commented:
OK, the latest changes work!
